core/src/mender-client.c

```diff
@@ -1165,17 +1009,30 @@ mender_client_update_work_function(void) {
         goto END;
     }
 
+    bool only_for_testing = true;
+
+    mender_log_info("Getting device type");
+
     /* Retrieve device type from artifact */
     const char *device_type_artifact = NULL;
     if (MENDER_OK != (ret = mender_artifact_get_device_type(mender_artifact_ctx, &device_type_artifact))) {
         mender_log_error("Unable to get device type from artifact");
-        mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_FAILURE);
-        if (mender_client_deployment_needs_set_pending_image) {
-            mender_flash_abort_deployment(mender_client_flash_handle);
+        if (true == only_for_testing) {
+            mender_log_error("Will be using PROJECT_NAME as device type for testing purposes");
+            char *device_type_test = malloc(strlen("PROJECT_NAME") + 1);
+            strcpy(device_type_test, "PROJECT_NAME");
```

Comment:
```suggestion
            char *device_type_test = strdup("PROJECT_NAME");
```

Reply:
Plus check that it was actually allocated as is not `NULL`.

Reply:
Thanks

Reply:
This code is removed. See above comment.